motionplan/motionPlanner.go

```diff
@@ -381,3 +382,81 @@ IK:
 	}
 	return orderedSolutions, nil
 }
+
+// CheckPlan checks if obstacles intersect the trajectory of the frame following the plan.
+func CheckPlan(
+	checkFrame frame.Frame,
+	plan []map[string][]frame.Input,
+	worldState *frame.WorldState,
+	fs frame.FrameSystem,
+	currentPosition spatialmath.Pose,
+	errorState spatialmath.Pose,
+) error {
+	// ensure that we can actually perform the check
+	if len(plan) < 1 {
+		return errors.New("plan must have at least one element")
+	}
+
+	// construct solverFrame
+	// Note that this requires all frames which move as part of the plan, to have an
+	// entry in the very first plan waypoint
+	sf, err := newSolverFrame(fs, checkFrame.Name(), frame.World, plan[0])
```

Comment:
I don't understand what it means for the world frame to be the goal. I also don't understand what the `seedMap` is & what it means for the first element of the plan to be the seedMap.

Reply:
> I don't understand what it means for the world frame to be the goal.

This means the goal is not relative to an arm/base/gantry. Instead it is relative to the world frame.
suppose we have a base whose parent is world. We also have a camera whose parent is base.
If the goal was relative to a base, for example, the coordinates of the goal would be different from the POV of the camera.